impress-org/givewp
# Add hash to export files to avoid collisions

## Description

When export files are created they have the same name, something like `give-payments.csv`. The problem is that due to the fact that exports are built via AJAX in multiple runs, it's possible for more than one export (from different users) to be happening simultaneously. In this case, the different exports would actually write to the same file.

This adds a hash to the file name on a per export basis to avoid this conflict. Note that it does not affect the final file name as downloaded by the user, as this is defined in the request header and isn't an issue. This only affects the file temporarily on the server for download.

## Affects

Exporting donations via Tools

## Pre-review Checklist




## Testing Instructions

Run an export and watch the `wp-contents/uploads` directory to see if the file temporarily generated has a hash in it. The downloaded file name should be as before and not contain a hash.
